### PR TITLE
Allow to use the MSVC 10.0 compiler from Windows SDK 7.1

### PR DESCRIPTION
**Pull request by tadeu**
Also allow to specify MSVC override via environment variable (CONDA_MSVC_OVERRIDE), but the specification in meta.yaml (build/msvc_compiler) still gets preference.

This is useful for Python stacks based on this alternative compiler, and also to not pollute all recipes with the msvc_compiler specification.

### CODE REVIEW

**Comment by tadeu** [comment]
I won't merge here, it's only for review before sending the PR to conda

**Comment by nicoddemus** [comment]
Looks good... are there any documentation to update though?

File: conda_build/windows.py [hunk 1]
```diff
@@ -66,6 +66,16 @@ def msvc_env_cmd(override=None):
         msvc_env_lines.append('set DISTUTILS_USE_SDK=1')
         msvc_env_lines.append('set MSSdk=1')
 
+    if version == '10.0-SDK':
+        # Special override to use the MSVC 2010 compiler (10.0) from Microsoft SDK 7.1
+        setenv = os.path.join(os.environ['ProgramFiles'], 'Microsoft SDKs', 'Windows', 'v7.1',
```
Comment (prusse-martin) [inline conda_build/windows.py]: 2010 compiler is not available in other (there is/will be another?) windows sdk?

Don't is better to get this path from registry? Maybe it is installed in a non standard place?
Key: HKEY_LOCAL_MACHINE
\SOFTWARE\Microsoft\Microsoft SDKs\Windows\v7.1\InstallationFolder

Comment (tadeu) [inline conda_build/windows.py]: > 2010 compiler is not available in other (there is/will be another?) windows sdk?

Each SDK have a different compiler version.

> Don't is better to get this path from registry? Maybe it is installed in a non standard place?
> Key: HKEY_LOCAL_MACHINE
> \SOFTWARE\Microsoft\Microsoft SDKs\Windows\v7.1\InstallationFolder

Ok, I'll try to do that

**Pull request closed without merging**